Cargo.toml

```diff
@@ -50,6 +50,7 @@ libgit2-sys = "=0.14.1"
 memchr = "2.1.3"
 opener = "0.5"
 os_info = "3.5.0"
+parking_lot = "0.12.1"
```

Comment:
Can you say why this needs parking_lot?
I'm generally reluctant to bring in new dependencies unless there is a strong reason. They increase build times, increase build and porting hazards, and potentially increase maintenance.

Reply:
From #11448

> It's both. I used std::sync::Mutex in an earlier iteration but found myself writing .unwrap() or .expect("thread didn't panic") which seemed worse than using parking_lot explicitly which is already part of the dependency tree.

Except I just looked and do not see `parking_lot` in the dependency tree

Reply:
Sorry, I think this might have come up in the other PR but wasn't addressed.

I chose it to avoid dealing with lock poisoning. Not using it either meant that `fn -> T` has to become `fn -> Result<T>` where ever the lock is obtained, or that I'd have to use `unwrap()` or `expect("no panic in thread")`, both of which seemed like the something to avoid.

Is there other options you see, or a choice you would make from the alternatives presented here?

Reply:
> Except I just looked and do not see parking_lot in the dependency tree

It's coming in with `gitoxide`.

Reply:
Typically for dealing with that I just call `.unwrap()`. The chance of having a poisoned lock is small if you keep the locked region small, and avoid other panics within it. And the consequences are usually small, unless you try to lock a mutex in a `drop()` method, which I would usually not recommend (though it looks like this PR does that). 

It is certainly not ideal, and there has been a lot of discussion on whether or not to actually try to introduce a different API. There is risk with the design of parking_lot since invariants may no longer hold if some locked region only partially completed its work.

I think it should probably be OK to bring it in if it will be required by gitoxide.